### PR TITLE
Refactoring Direct DB query in orders_with_charge_id_from_charge_ids 

### LINKED ISSUE #4151: Refactor the direct post table queries found in the `WC_Payments_DB` class to use `wc_get_orders()`/`WC_Order_Query`
### Description

With [Custom Order tables coming to WooCommerce Core](https://developer.woocommerce.com/2022/01/17/the-plan-for-the-woocommerce-custom-order-table/), direct DB queries that fetch orders by post meta will need to be reworked/updated to work with orders stored in custom tables.

There are a few functions in the `WC_Payments_DB` class which are querying order IDs and order meta that will need to be updated:
 - [ ] [`orders_with_charge_id_from_charge_ids()`](https://github.com/Automattic/woocommerce-payments/blob/ca0aa7a9160647e3612bffc8b9e0f935bea3acd1/includes/class-wc-payments-db.php#L46-L52) - Retrieve orders from the DB using a list of corresponding Stripe charge IDs.
 - [ ] [`order_id_from_meta_key_value()`](https://github.com/Automattic/woocommerce-payments/blob/ca0aa7a9160647e3612bffc8b9e0f935bea3acd1/includes/class-wc-payments-db.php#L93-L99) - Retrieve the latest order ID from the DB using a meta key value pair.

This issue is to look at replacing these queries with `wc_get_orders()`/`WC_Order_Query` alternatives.

### Dev notes

There's a wiki on how to search for orders using custom meta here: https://github.com/woocommerce/woocommerce/wiki/wc_get_orders-and-WC_Order_Query#adding-custom-parameter-support

### Additional context

**Project thread:** pdjTHR-Xp-p2
**Exploration p2:** pdjTHR-YX-p2

### PR DESCRIPTION
**Pull request by jessy-p**
Fixes #4151

#### Changes proposed in this Pull Request

Replaced the direct DB query in `orders_with_charge_id_from_charge_ids()` with `wc_get_orders`

#### Testing instructions
The function `orders_with_charge_id_from_charge_ids` is used in `WC_Payments_API_Client::list_transactions` which is used in the REST API `/transactions`. This API is used on the Transactions listing page.

1) Place an order, and ensure its captured (if Capture Later is enabled)
2) Sync Transactions `wp wcpay sync_transactions_for <account_id>`
3) Go to **Payments** > **Transactions** and check that the transaction is listed.
4) The above steps should be tested under all the following conditions:
    -  No COT
    -  COT, using orders table and no sync
    -  COT, using posts table and no sync
    -  COT, using order table with sync
    -  COT, using posts table with sync

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/class-wc-payments-db.php [hunk 1]
```diff
@@ -38,28 +38,26 @@ public function order_from_charge_id( $charge_id ) {
...
+			]
 		);
 
 		return array_map(
-			function ( stdClass $row ) : array {
+			function ( WC_Order $order ) : array {
 				return [
-					'order'     => $this->order_from_order_id( $row->order_id ),
-					'charge_id' => $row->charge_id,
+					'order'     => $this->order_from_order_id( (string) $order->get_id() ),
```
Comment (deepakpathania) [inline includes/class-wc-payments-db.php:54]: Shouldn't we be getting the `order` itself from `wc_get_orders` in the first place? Why do we need to call this by extracting the id from it?

File: includes/class-wc-payments-db.php [hunk 2]
```diff
@@ -38,28 +38,26 @@ public function order_from_charge_id( $charge_id ) {
 	 * @return array[]
```
Comment (deepakpathania) [inline includes/class-wc-payments-db.php:38]: Should we be more specific about the return type here for easier reference? I feel it helps with testing.

**Review by deepakpathania — COMMENTED** [review]
Tests well for COT enabled and disabled scenarios, left a few comments.

File: includes/class-wc-payments-db.php (hunk 1 above)
Comment (jessy-p) [reply includes/class-wc-payments-db.php:54]: Sorry, missed that. Fixed now.

File: includes/class-wc-payments-db.php (hunk 2 above)
Comment (deepakpathania) [reply includes/class-wc-payments-db.php:38]: Discovered that is pretty complex to represent an array of objects with mixed types in the phpdoc directly, something for a wider discussion later, resolving this for now.

**Review by deepakpathania — APPROVED** [review]
LGTM now. ✅

**Pull request merged**